kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeDownloadPanel.js

```diff
@@ -25,70 +29,71 @@ define([
             objName: null,
             downloadSpecCache: null, // {'lastUpdateTime': <millisec.>, 'types': {<type>: <spec>}}
         },
-        token: null,
-        type: null, // Type of workspace object to show downloaders for
-        objId: null,
         loadingImage: Config.get('loading_gif'),
         shockURL: Config.url('shock'),
-        // TODO: remove data_import_export code
         exportURL: Config.url('data_import_export'),
-        useDynamicDownloadSupport: false,
         nmsURL: Config.url('narrative_method_store'),
         eeURL: Config.url('execution_engine2'),
-        srvWizURL: Config.url('service_wizard'),
         timer: null,
-        downloadSpecCache: null, // {'lastUpdateTime': <millisec.>, 'types': {<type>: <spec>}}
 
         init: function (options) {
             this._super(options);
-            this.token = this.options.token;
-            this.type = this.options.type;
-            this.objId = this.options.objId;
-            this.objName = this.options.objName;
-            this.ref = this.options.ref;
+            Object.assign(this, options);
             if (!this.objId) {
-                const refPathItems = this.ref.split(';');
-                this.objId = refPathItems[refPathItems.length - 1].trim().split('/')[1];
+                this.objId = this.ref.split(';').pop().trim().split('/')[1];
```

Comment:
would be useful to have a comment here to say what this is doing so whoever is reading it doesn't have to engage their brain too much...

Reply:
Yup, good idea. Added.